src/main/java/com/csfrequency/firebase/devicestore/FirebaseDeviceStore.java

```diff
@@ -287,12 +225,18 @@ public void onReceive(Context context, Intent intent) {
             }
 
             String token = intent.getStringExtra(FDSMessagingService.TOKEN_EXTRA);
-            // If the token has changed, then update it
-            if ((token == null && currentToken != null || !token.equals(currentToken)) && currentUser != null) {
+
+            // If there's no current user, just update the cached token
+            if (currentUser == null) {
+                currentToken = token;
+                return;
+            }
+
+            // If the token has changed, then update in Firestore and update the cached token
+            if ((token == null && currentToken != null || !token.equals(currentToken))) {
```

Comment:
Nit: Extra parenthesis.

If you haven't, do you mind running `google-java-format` over your sources? This is the code formatter we use in our GitHub repos.

Reply:
I've now set up and run `google-java-format`